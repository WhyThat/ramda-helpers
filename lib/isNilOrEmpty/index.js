import { isNil, isEmpty, anyPass } from 'ramda';

/**
 * Test if value is NilOrEmpty
 * @name isNilOrEmpty
 * @memberof RamdaHelper#
 * @see isPathSatisfied if you want to test nested object
 * @param {*} val
 * @returns {bool}
 * @example
 * isNilOrEmpty([]) //=> true
 * isNilOrEmpty({}) //=> true
 * isNilOrEmpty('') //=> true
 * isNilOrEmpty(null) //=> true
 * isNilOrEmpty(false) //=> false
 * isNilOrEmpty('test') //=> false
 * isNilOrEmpty(['test']) //=> false
 * isNilOrEmpty({ a: 1 }) //=> false
 */
export default anyPass([isEmpty, isNil]);
