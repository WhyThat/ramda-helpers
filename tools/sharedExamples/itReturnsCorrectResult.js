export default (result) => {
  test('returns correct result', () => {
    expect(result).toMatchSnapshot();
  });
};
