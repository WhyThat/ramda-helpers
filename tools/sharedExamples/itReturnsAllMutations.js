import itReturnsCorrectResult from './itReturnsCorrectResult';

export default (fn, mutations) => {
  mutations.forEach((mutation, index) => {
    const name = mutation.name || `mutations ${index}`;

    describe(name, () => {
      itReturnsCorrectResult(fn(mutation.props));
    });
  });
};
