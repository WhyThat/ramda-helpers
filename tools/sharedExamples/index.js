
import itReturnsAllMutations from './itReturnsAllMutations';
import itReturnsCorrectResult from './itReturnsCorrectResult';

export {
  itReturnsAllMutations,
  itReturnsCorrectResult,
};
