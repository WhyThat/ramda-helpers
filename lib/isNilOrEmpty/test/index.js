import { itReturnsCorrectResult } from '../../../tools/sharedExamples';
import isNilOrEmpty from '../';

describe('isNilOrEmpty', () => {
  const mutations = [{
    name: 'With null',
    props: null,
  }, {
    name: 'With undefined',
    props: undefined,
  }, {
    name: 'With empty string',
    props: '',
  }, {
    name: 'With empty array',
    props: [],
  }, {
    name: 'With empty object',
    props: {},
  }, {
    name: 'With  an no empty string',
    props: 'el',
  }, {
    name: 'With no empty array',
    props: ['el'],
  }, {
    name: 'With no empty object',
    props: {
      el: '',
    },
  }, {
    name: 'With false',
    props: false,
  }];

  mutations.forEach((mutation) => {
    describe(mutation.name, () => {
      itReturnsCorrectResult(isNilOrEmpty(mutation.props));
    });
  });
});

