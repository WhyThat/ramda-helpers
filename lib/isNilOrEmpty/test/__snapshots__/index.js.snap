// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isNilOrEmpty With  an no empty string returns correct result 1`] = `false`;

exports[`isNilOrEmpty With empty array returns correct result 1`] = `true`;

exports[`isNilOrEmpty With empty object returns correct result 1`] = `true`;

exports[`isNilOrEmpty With empty string returns correct result 1`] = `true`;

exports[`isNilOrEmpty With false returns correct result 1`] = `false`;

exports[`isNilOrEmpty With no empty array returns correct result 1`] = `false`;

exports[`isNilOrEmpty With no empty object returns correct result 1`] = `false`;

exports[`isNilOrEmpty With null returns correct result 1`] = `true`;

exports[`isNilOrEmpty With undefined returns correct result 1`] = `true`;
